include/simdjson/document.h

```diff
@@ -93,16 +93,18 @@ namespace simdjson::dom {
  * This is the type it is most easily cast to with get<>.
  */
 enum class element_type {
-  ARRAY,     ///< dom::array
-  OBJECT,    ///< dom::object
-  INT64,     ///< int64_t
-  UINT64,    ///< uint64_t: any integer that fits in uint64_t but *not* int64_t
-  DOUBLE,    ///< double: Any number with a "." or "e" that fits in double.
-  STRING,    ///< std::string_view
-  BOOL,      ///< bool
-  NULL_VALUE ///< null
+  ARRAY = 7,     ///< dom::array
+  OBJECT = 11,    ///< dom::object
+  INT64 = 10,     ///< int64_t
+  UINT64 = 5,    ///< uint64_t: any integer that fits in uint64_t but *not* int64_t
+  DOUBLE = 2,    ///< double: Any number with a "." or "e" that fits in double.
+  STRING = 6,    ///< std::string_view
+  BOOL = 4,      ///< bool
+  NULL_VALUE = 12 ///< null
 };
 
+/* We choose the integers 7, 11, and so forth in element_type so that the matching tape values (", {,..,) are mapped to them using the x -> x % 14 function. */
```

Comment:
Can you move this note up to the beginning of element_type (I guess first line before the array)? I didn't see this line at first because it was below and separated by a space, so I stared at those numbers for a while trying to make sense of them.

What happens if this becomes STRING = tape_type::STRING, etc., and we make BOOL = tape_type::TRUE_VALUE? ?Then instead of x % 14 the mapping function could be `x == tape_type::FALSE_VALUE ? tape_type::BOOL : tape_type`, which ... presumably? would use a CMOV.

Come to think, it's likely slower than x % 14, with 2 instructions instead of 1, so ... never mind.

Reply:
Oh never mind, it's not likely slower given that we had to use some supermath to make it work :) Might be worth trying CMOV ...

Reply:
I Iike the cmov idea. I will test it out.

Reply:
I think that the cmov works well. It is definitively much simpler than my acrobatics.